### DISCUSSION TITLE
Installing Subpackage Installs Full AutoRA

### DISCUSSION THREAD

**Discussion by chadcwilliams** [AutoResearch/autora#410] [Core Contributions] [answered]
Ben just helped me through releasing the 3 experimentalist repo's I parsed from AutoRA. This went very well and then we modified the AutoRA Restructuring branch to include them as dependencies. All worked out well with this process.

Where an issue arose was when I then tried to install the experimentalist package. I built a novel environment and then as per quickstart installed one of the experimentalists I just released (this is the first command in the new virtual environment, so I didn't install anything else): 

`pip install -U autora["experimentalist-model-disagreement-sampler"]`

The consequent of this seems to be to install Autora 2.4.1 (not the restructured package) and does output a warning:
`WARNING: autora 2.4.1 does not provide the extra 'experimentalist-model-disagreement-sampler'`

Maybe I am installing the restructured package wrong?

**Comment by benwandrew** [comment]
maybe until the restructured mother package is no longer a branch (ie, until it becomes `main`) the command requires some additional flag? per the pre-release PyPI page (https://pypi.org/project/autora/3.0.0a14/), perhaps the following?

`autora==3.0.0a14`

@younesStrittmatter @gtdang thoughts?

**Reply by benwandrew** [reply]
looks like `--pre` might be the requisite flag?

https://pip.pypa.io/en/stable/cli/pip_install.html#:~:text=the%20pip%20install%20command%20also%20supports%20a%20%2D%2Dpre%20flag%20that%20enables%20installation%20of%20pre%2Dreleases%20and%20development%20releases

`--pre` includes pre-releases in the search for the latest version to install, and since we shouldn’t be releasing a new version of `autora` before restructuring is complete (?), that flag should suffice? alternatively, can maybe specify a particular pre-release using `autora==` per above.

so, `pip install --pre -U autora["experimentalist-model-disagreement-sampler"]` should work?

**Reply by gtdang** [reply, marked as answer]
I think the pre flag should work to get the latest pre-release. If you wanted to specify exact version try 
`pip install -U "autora[experimentalist-model-disagreement-sampler]==3.0.0a14"`

When supplying version numbers and extras I think the whole thing should be in quotes like the example above.
